automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/security/AuthDomainSpec.scala

```diff
@@ -28,169 +30,468 @@ class AuthDomainSpec extends FreeSpec /*with ParallelTestExecution*/ with Matche
-        withGroup("AuthDomainSpec") { authDomainName =>
-          withWorkspace(projectName, "AuthDomainSpec", Some(authDomainName)) { workspaceName =>
-            signIn(Config.Users.ron)
+              val listPage = signIn(Config.Users.george)
+              val summaryPage = listPage.openWorkspaceDetails(projectName, workspaceName)
 
-            val workspaceSummaryPage = new WorkspaceSummaryPage(projectName, workspaceName)
-            go to workspaceSummaryPage
-            workspaceSummaryPage.awaitLoaded()
-            workspaceSummaryPage.ui.readError() should include(projectName)
-            workspaceSummaryPage.ui.readError() should include(workspaceName)
-            workspaceSummaryPage.ui.readError() should include("does not exist")
+              val cloneWorkspaceName = workspaceName + "_clone"
+              val cloneModal = summaryPage.ui.clickCloneButton()
+              cloneModal.ui.readLockedAuthDomainGroups() should contain(authDomainName)
+
+              register cleanUp {
+                api.workspaces.delete(projectName, cloneWorkspaceName)(AuthTokens.george)
+              }
+              cloneModal.cloneWorkspace(projectName, cloneWorkspaceName)
+              cloneModal.cloneWorkspaceWait()
+              summaryPage.cloneWorkspaceWait()
+
+              val cloneSummaryPage = new WorkspaceSummaryPage(projectName, cloneWorkspaceName).awaitLoaded()
+              cloneSummaryPage.ui.readWorkspaceName should be(cloneWorkspaceName)
+              cloneSummaryPage.ui.readAuthDomainGroups should include(authDomainName)
+            }
           }
         }
       }
 
-      "should not be visible" in withWebDriver { implicit driver =>
-        withGroup("AuthDomainSpec") { authDomainName =>
-          withWorkspace(projectName, "AuthDomainSpec_share", Some(authDomainName)) { workspaceName =>
-            val listPage = signIn(Config.Users.ron)
-            listPage.filter(workspaceName)
-            listPage.ui.hasWorkspace(projectName, workspaceName) shouldEqual false
+      "when the user is not inside of the group" - {
+        "when the workspace is shared with them" - {
+          "can be seen but is not accessible" in withWebDriver { implicit driver =>
+            withGroup("AuthDomainSpec") { authDomainName =>
+              withWorkspace(projectName, "AuthDomainSpec_reject", Set(authDomainName)) { workspaceName =>
+                api.workspaces.updateAcl(projectName, workspaceName, Config.Users.ron.email, WorkspaceAccessLevel.Reader)
+
+                val workspaceListPage = signIn(Config.Users.ron)
+                workspaceListPage.filter(workspaceName)
+                workspaceListPage.ui.looksRestricted(projectName, workspaceName) shouldEqual true
+
+                workspaceListPage.ui.clickWorkspaceInList(projectName, workspaceName)
+                // micro-sleep just long enough to let the app navigate elsewhere if it's going to, which it shouldn't in this case
+                Thread sleep 500
```

Comment:
This was written before the "request access" modal existed. I agree that we should update it to be a better test.

I agree that we should only use `sleep` as an absolute last resort and we should always question its usage when we see it.